core/cmd/shell.go

```diff
@@ -969,14 +969,14 @@ func NewFileAPIInitializer(file string) APIInitializer {
 	return fileAPIInitializer{file: file}
 }
 
-func (f fileAPIInitializer) Initialize(orm sessions.ORM, lggr logger.Logger) (sessions.User, error) {
+func (f fileAPIInitializer) Initialize(orm sessions.UserManager, lggr logger.Logger) (sessions.User, error) {
 	request, err := credentialsFromFile(f.file, lggr)
 	if err != nil {
 		return sessions.User{}, err
 	}
 
 	// Load list of users to determine which to assume, or if a user needs to be created
-	dbUsers, err := orm.ListUsers()
+	dbUsers, err := orm.LocalAdminListUsers()
```

Comment:
@CL-Andrew I'm likely missing some context around this, but this feels a bit smelly to me. If we're delegating our auth to an external LDAP server, why do we sometimes need to fallback to a local server?

Could we solve the expansion of the API by having a local auth implementation, and then an LDAP implementation that wraps the local auth implementation, augmenting auth requests with external calls as needed? And in this particular case where we seem to need the local auth implementation, we pass that in rather than the wrapped LDAP implementation. I may be way off base here though.

Reply:
Local auth is used if an LDAP server becomes unavailable or needs settings changed.

If auth is fully delegated, then this becomes a challenging problem to solve.

Reply:
No matter the backend authentication provider is, a local admin user will always be required for the initial setup and and local admin CLI commands. I'm not sure how well that would fit as all of the non admin functions in the new UserManager interface are exclusive to the provider implementing it. Another effect of this is that the node is still accessible through the local CLI if there is an issue with the authentication server, or it needs updated. Do you think we can improve the comments or organization here?

Reply:
> No matter the backend authentication provider is, a local admin user will always be required for the initial setup and and local admin CLI commands.

Hmmm... this feels like further evidence to me that we should be using the local auth implementation here rather than extending the auth server with additional admin specific methods.

Reply:
I see your point, that makes sense. What do you think about extending the top level ChainlinkApplication struct with a new `authProvider UserManager` field, alongside the existing and currently conflated `sessionORM sessions.UserManager` field. In case of local auth, authProvider will still be initialized and stored identical to sessionORM, and call sites requiring the local auth can reference the top level strut sessionORM. I'll push a rewrite?

Reply:
Yes, I think that works!

Reply:
Okay! Updated now, PTAL

Reply:
LGTM!